src/helpers/buffers/unordered_receiver.rs

```diff
@@ -284,6 +314,19 @@ where
             _marker: PhantomData,
         }
     }
+
+    /// Returns `true` when this receiver is closed. Closed means the underlying stream is done and
+    /// there is no more data inside receiver's buffers.
+    ///
+    /// Calling `poll_next` on closed receivers will result in [`EOS`] error.
+    ///
+    /// [`EOS`]: crate::helpers::Error::EndOfStream
+    pub fn is_closed(&self) -> bool {
+        // If this function is ever called on the hot path, consider caching closed status.
+        // Closed streams cannot move back to open.
+        let inner = self.inner.lock().unwrap();
+        inner.stream.is_done() && inner.spare.is_empty()
```

Comment:
Consider what might happen if you have spare bytes that are too small for a single message.  I think that tends to end up with an error, but you won't return `is_closed() == true` in that case.

I think that you want to flush spare in that case.

Reply:
I believe now it is handled - next poll will return an error and stream will be properly closed